lib/daru/plotting/dataframe.rb

```diff
@@ -89,6 +91,21 @@ def plot_with_category opts
         plot.show
       end
 
+      SHAPES = ['circle','triangle-up', 'diamond', 'square', 'triangle-down', 'cross']
+      def get_shape
+        return to_enum(:get_shape) unless block_given?
+        loop do
```

Comment:
wont the presence of this loop make the each interation happen infinitely? `#each` will handle what you want to do, if I'm not missing something.